ee/agent/knapsack/knapsack.go

```diff
@@ -51,7 +51,7 @@ func New(stores map[storage.Store]types.KVStore, flags types.Flags, db *bbolt.DB
 	}
 
 	if k.slogger != nil {
-		k.slogger.Logger = k.slogger.Logger.With("launcher_run_id", k.launcherRunId)
+		*k.slogger.Logger = *k.slogger.Logger.With("launcher_run_id", k.launcherRunId)
```

Comment:
This spot feels like it breaks an abstraction. Somehow it feels wrong for knapsack to be reaching into slogger.Logger. But 🤷 not the end of the world

Reply:
I agree, it's a weird spot. Knapsack needs access to the multislogger to be able to add handlers. We also want the launcher_run_id to be universally included in all logs ..... hmm maybe multislogger should just add it, then knapsack can stop reaching in there

Reply:
I moved this down to the multislogger, now knapsack is staying in it's lane and not reaching down into multislogger's guts